python_modules/dagster/dagster/core/storage/runs/migration.py

```diff
@@ -51,3 +54,19 @@ def migrate_run_partition(storage, print_fn=None):
             continue
 
         storage.add_run_tags(run.run_id, run.tags)
+
+
+def migrate_run_start_end(storage, print_fn=None):
+    """
+    Utility method that updates the start and end times of historical runs using the completed event log.
+    """
+
+    if print_fn:
+        print_fn("Querying run and event log storage.")
+
+    for run in chunked_run_iterator(storage, print_fn):
+        run_record = storage.get_run_records(PipelineRunsFilter(run_ids=[run.run_id]))[0]
+        if run_record.start_time:
+            continue
+
+        storage.add_run_stats(run.run_id)
```

Comment:
What do you think about putting most of this migration functionality in a migration function here, instead of on the storage object itself?  Might make it more obvious when we can delete it / when we are using it.

Added benefit that we can assert that the storage is an instance of the SqlRunStorage, which I think is the only kind of run storage that uses data migrations.  Then we don't have to put this on the base / in-memory implementations